### PR TITLE
feat: Front Page: Move Search Function to Top Bar 

### LINKED ISSUE #5927: Front Page: Move Search Function to Top Bar
Please move the search feature to the top bar and substitute "Browse Events" with "Search Events". Implement it as follows:
* Implement the search in the place of "Browse Events" 
* Use a similar style and show "Search Events"
* Expand the search box similar to GitHub when the user enters something
* Delete the main search bar below the header
* Show the top search bar on the same pages as Browse Events at the moments (do not change anything here)

Screenshot from 2020-12-06 01-51-47

### PR DESCRIPTION
**Pull request by daretobedifferent18**
Fixes #5927 

#### Short description of what this resolves:
Move the search feature to the top bar and substitute "Browse Events" with "Search Events"

#### Changes proposed in this pull request:
  - [x] Implement the search in the place of "Browse Events"
  - [x] Use a similar style and show "Search Events"
  - [x] Expand the search box similar to GitHub when the user enters something
  - [x] Delete the main search bar below the header
  - [x] Show the top search bar on the same pages as Browse Events at the moments (do not change anything here)
  - [x] Implement search box on Mobile view

#### Screenshot

Desktop
Screenshot from 2020-12-13 03-22-35

Mobile(sideBar) | Mobile(Navbar Search Icon) | Mobile(Navbar Search Icon(on click))
----------- | ----------- | ----------
Screen Shot 2020-12-13 at 03 21 44 | !Screen Shot 2020-12-13 at 23 19 24 | Screen Shot 2020-12-13 at 23 19 27

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mariobehling** [comment]
Thank you. Please change the following:
* Switch search one position to the right
* Do not keep a straight line to the left
* Do not keep a border for the search box
* Add a search lens
* Expand search box length to the left when the user clicks into it

Screenshot from 2020-12-12 01-40-57

**Comment by daretobedifferent18** [comment]
@mariobehling Pls have a look

**Comment by iamareebjamal** [comment]
Please use #fafafa as bg color. Make default size of search bar smaller and don't expand it too much. GitHub expands it to show suggestions, we are not doing that currently.

**Comment by iamareebjamal** [comment]
And it should also display and work on mobile

**Comment by iamareebjamal** [comment]
Great, however I think mobile view should also contain the search bar in NavBar, or people won't notice it

**Comment by daretobedifferent18** [comment]
Then pls let me know where should I implement that. Bcz no space is in navbar

**Comment by iamareebjamal** [comment]
image

Lot of space

**Comment by iamareebjamal** [comment]
If there is no space, do what eventbrite does

**Comment by mariobehling** [comment]
Yes, please implement it in the top menu bar. Thanks

**Comment by mariobehling** [comment]
Please also:
* Make the expanded version of the search box 20px longer
* Make the lens icon in the same color as the "Search Events" font
Screenshot from 2020-12-13 08-47-37

**Comment by daretobedifferent18** [comment]
@iamareebjamal Pls have a look

**Comment by iamareebjamal** [comment]
Search icon still black. Can't click on cross till I type something

**Comment by daretobedifferent18** [comment]
> Search icon still black. Can't click on cross till I type something

cross icon is working for me

**Comment by daretobedifferent18** [comment]
@iamareebjamal I have made the changes and cross icon is working

**Review by sachinchauhan2889 — CHANGES_REQUESTED** [review]
look nice to me. 
only github actions / lint annotations needs to fix.

**Comment by daretobedifferent18** [comment]
> look nice to me.
> only github actions / lint annotations needs to fix.

lint is not giving any error

File: app/components/nav-bar.js [hunk 1]
```diff
@@ -9,6 +9,24 @@ export default class NavBar extends Component {
...
+  handleKeyPress() {
+    if (event.code === 'Enter') {
+      this.sendAction('search');
+      document.querySelector('#mobile-bar').classList.remove('show-bar');
+    }
+  }
+
+  @action
+  toggleSearchBar() {
+    document.querySelector('#mobile-bar').classList.toggle('show-bar');
```
Comment (iamareebjamal) [inline app/components/nav-bar.js:27]: Why not using ember for this? Merging this as it has been too long, but please create a follow up PR to change DOM manipulation to ember

**Review by iamareebjamal — APPROVED** [review]

**Comment by iamareebjamal** [comment]
Great job @daretobedifferent18 

Please follow up with a PR with minor change I mentioned

**Comment by daretobedifferent18** [comment]
@iamareebjamal also should I do #6017 with it bcz I think it will be easy to do it together bring search bar there as well

**Comment by iamareebjamal** [comment]
Yes

**Pull request merged**